include/wlr/types/wlr_text_input.h

```diff
@@ -0,0 +1,77 @@
+#ifndef WLR_TYPES_WLR_TEXT_INPUT_H
```

Comment:
This file should have a `_v3` suffix (alongside with all structs).

Ah, and you need to add the `WLR_USE_UNSTABLE` stuff too at the beginning of all new header files.